tests/python/test_local_atomics.py

```diff
@@ -29,3 +29,155 @@ def func():
 
     func()
     assert A[None] == 45
+
+
+@ti.all_archs
+def test_explicit_local_atomic_sub():
+    A = ti.var(ti.f32, shape=())
+
+    @ti.kernel
+    def func():
+        a = 0
+        for i in range(10):
+            ti.atomic_sub(a, i)
+        A[None] = a
+
+    func()
+    assert A[None] == -45
+
+
+@ti.all_archs
+def test_implicit_local_atomic_sub():
+    A = ti.var(ti.f32, shape=())
+
+    @ti.kernel
+    def func():
+        a = 0
+        for i in range(10):
+            a -= i
+        A[None] = a
+
+    func()
+    assert A[None] == -45
+
+
+@ti.all_archs
+def test_explicit_local_atomic_min():
+    A = ti.var(ti.f32, shape=())
+
+    @ti.kernel
+    def func():
+        a = 1000
+        for i in range(10):
+            ti.atomic_min(a, i)
+        A[None] = a
+
+    func()
+    assert A[None] == 0
+
+
+@ti.all_archs
+def test_explicit_local_atomic_max():
+    A = ti.var(ti.f32, shape=())
+
+    @ti.kernel
+    def func():
+        a = -1000
+        for i in range(10):
+            ti.atomic_max(a, i)
+        A[None] = a
+
+    func()
+    assert A[None] == 9
+
+
+@ti.all_archs
```

Comment:
Great work! I think @k-ye and me will add these atomic ops to metal and opengl later so they'll pass this all_archs.

Reply:
Yep, created #632